imgui-core/imgui/imconfig.h

```diff
@@ -0,0 +1,134 @@
+//-----------------------------------------------------------------------------
+// COMPILE-TIME OPTIONS FOR DEAR IMGUI
+// Runtime options (clipboard callbacks, enabling various features, etc.) can generally be set via the ImGuiIO structure.
+// You can use ImGui::SetAllocatorFunctions() before calling ImGui::CreateContext() to rewire memory allocation functions.
+//-----------------------------------------------------------------------------
+// A) You may edit imconfig.h (and not overwrite it when updating Dear ImGui, or maintain a patch/rebased branch with your modifications to it)
+// B) or '#define IMGUI_USER_CONFIG "my_imgui_config.h"' in your project and then add directives in your own file without touching this template.
```

Comment:
> I fear that maintenance of that file will be fragile.

Yes, this is the case with all modifications to the upstream but oftentimes there is no choice. The way we normally do it in project that we package is to (1) have an `.orig` symlinkg to the original next to the modified version and (2) mention that this file need to be re-patched during upstream upgrade in the `README-DEV` file that we create in the root of a repository (we also put any other packaging-related notes there).

Reply:
Alright I will then add a `README-DEV` file with according notes. Symlink will also be added.

@Klaim I see your point, that it can be hairy for package users to inject the configuration they want. However, afaik the original imgui package did not provide any further configuration options for this header. So I would see the feature to provide advanced configuration options in a different PR and not in this one.

Reply:
> However, afaik the original imgui package did not provide any further configuration options for this header. So I would see the feature to provide advanced configuration options in a different PR and not in this one.

Yes, that's  what I meant by "in the future".

Note that IMGUI historically didn't support any (meta-)build-system, including CMake: you had to copy-paste the sources and tweak them. They still want users to do that, which is why there is no mechanism provided by default from a CMake point of view. It would still be an improvement to generate that file in the context of `build2` (as a luxury though) because of the ability to work on multiple configurations at the same time.
Anyway, this can be noted in an issue after this PR is finihed, 👍🏽

Reply:
Added `README-DEV` and raised a new ticket concerning `imconfig.h`.

https://github.com/build2-packaging/imgui/issues/3